src/main/scala/akka/persistence/cassandra/query/EventsByTagPublisher.scala

```diff
@@ -135,6 +143,9 @@ private[query] class EventsByTagPublisher(
 
     case Cancel =>
       context.stop(self)
+
+    case tagWritten: String if tagWritten == tag =>
+      self ! Continue
```

Comment:
We dealt with over-polling in our earlier [query implementation](https://github.com/jypma/akka-persistence-cassandra-query) by splitting every `getEventsByTag` call into two phases: reading from past, and forwarding real-time events. All ongoing `getEventsByTag` calls that had caught up with real time were then driven by a shared cassandra polling process, since they're all going to get the same events anyways. I haven't investigated into what would be involved with trying to attempt something like that here.